pkg/server/registry/images/pull.go

```diff
@@ -155,7 +155,7 @@ func (i *ImagePull) ImagePull(ctx context.Context, namespace, imageName string)
 
 		// don't write error to chan because it already gets sent to the progress chan by remote.WriteIndex()
 		if err = remote.WriteIndex(repo.Digest(hash.Hex), index, opts...); err == nil {
-			if err := i.clientFactory.Namespace(namespace).ImageTag(ctx, hash.Hex, imageName); err != nil {
+			if _, err := i.clientFactory.Namespace(namespace).ImageCreate(ctx, hash.Hex, imageName); err != nil {
```

Comment:
You should change this one to creating ImageInstance now. Since this is server side, we can safely create ImageInstances and @ibuildthecloud doesn't want anything directly creating Image, just ImageInstance.

The client side creates (like in cli/build.go), we have to keep as Image until darren's refactor is done

Reply:
I'm a little confused on this one, do we have an image at this point to where we can call i.client.create(ctx,image) which would make the image instance? @cjellick

Reply:
addressed in slack. TL;DR
```
+			img := &v1.ImageInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      imageName,
+					Namespace: namespace,
+				},
+				Digest: hash.Hex,
+			}
+			if err := i.client.Create(ctx, img); err != nil {
```